### PR TITLE
Minor training improvements

### PR DESCRIPTION
**Pull request by matt-gardner**
There are a bunch of unrelated things in here.  I tried to split them out into their own commits, so you can look at the commit log and click on individual commits if you want a better picture of what's in here and why.

### CODE REVIEW

File: allennlp/nn/util.py [hunk 1]
```diff
@@ -113,6 +114,10 @@ def arrays_to_variables(data_structure: Dict[str, Union[dict, numpy.ndarray]],
         using this function. This is useful during inference for passing
         tensors representing a single example to a Pytorch model which
         would otherwise not have a batch dimension.
+    for_training : ``bool``, optional (default = ``True``)
```
Comment (DeNeutoy) [inline allennlp/nn/util.py]: Can you add this flag to all of the inference methods on the models too?

Comment (DeNeutoy) [inline allennlp/nn/util.py]: Also I think it would be more accurate to say "this makes inference slightly faster and considerably more memory efficient".

Comment (DeNeutoy) [inline allennlp/nn/util.py]: Also, maybe it's worth just sticking to the pytorch name for this flag?

Comment (matt-gardner) [inline allennlp/nn/util.py]: The trouble with using `volatile` as the name here is that it has no relation to anything.  `volatile` to me is about thread safety, not about whether the tensor will allow gradient computation.  So `volatile` is an incredibly unhelpful name for this parameter.  On the other hand, the trouble with changing it is that if you're looking for how to pass the `volatile=True` flag in pytorch to the variable creation code, it's not as easy to find what you're looking for.  That's why I added in the comment.

Comment (DeNeutoy) [inline allennlp/nn/util.py]: Sure, just a suggestion.

**Pull request merged**